src/types.ts

```diff
@@ -187,11 +191,59 @@ export interface Client extends EvaluationLifeCycle, Features {
   readonly version?: string;
 }
 
-export type HookContext = {
-  // TODO: implement with hooks
-};
+export type HookHints = Readonly<Record<string, unknown>>;
+
+export interface HookContext<T extends FlagValue = FlagValue> {
+  readonly flagKey: string;
+  readonly defaultValue: T;
+  readonly flagType: FlagType;
+  readonly context: Readonly<EvaluationContext>;
+  readonly client: Client;
+  readonly provider: Provider;
+}
+
+export interface BeforeHookContext extends HookContext {
+  context: EvaluationContext;
+}
```

Comment:
The `before` hook now has a slightly different type to denote the fact that the context is mutable only in this case (other hooks can't mutate context, per spec).